tests/e2e/script_test.go

```diff
@@ -112,13 +142,41 @@ func TestE2EScript(t *testing.T) {
 						// The object probably still exists (that's probably
 						// why we're using DELETE-NO-WAIT), so export the
 						// resource and the kube object.
+					case "CHECK-LOG":
```

Comment:
I'm still not very clear what CHECK-LOG is doing...

Reply:
parking some thoughts here

> what CHECK-LOG is doing...

Would there be a better name for it?

---

I think there's still some value in having a mechanism to check this scenario: "when CC/ CCC start reconciling again via an APPLY, let's check that the logs show KCC resources start getting reconciled onto the cloud provider again". IIRC, if it's sufficient to check the logs for when we apply CC/ CCC we could get away with just having the golden checks. But at present apply CC/ CCC is "instantaneous" and we move on without recording any http events. Wdyt ?

---

IIRC, we did chat about forcing the reconcile on the KCC object (ArtifactRegistryRepository here). That approach still requires some timing/ waiting that for the logs to get captured as part of apply a CC/ CCC resource that turns on `Reconciling`. 

On a separate note, I think we should add an option to turn off or cap the jiiter as that can add an extra level of non determinism here. (i.e. we wait on the object to get reconciled or worse case the jiiter pushes out past the time window that we are waiting for).

Reply:
Ah OK, so I think I understand now.

Let's see what happens with the jitter.  I imagine we don't want to waiting around for a re-reconcile in general (it'll make our tests slow), so we will want to trigger reconciles.  We may want to make the re-reconcile interval _longer_, in that if our test happened to go slowly, we might see a "random" re-reconcile from another object when we aren't expecting it.  But I propose we tackle that when we start to see it.

So I think CHECK-LOG could be "WAIT-FOR-HTTP-REQUEST" or something along those lines.  And then I think ideally it should not have an object at all, we should just use the "APPLY" task.  Now I think the problem with that is that we need to find a change for each object that is reflected in metadata.generation, and I'm not sure if labels are reflected in metadata.generation (which is its own problem; we know we need to rethink labels).

So I'm OK to have this now that I understand what's going on :-), but let's work in this direction and see what's possible.